dashboard/app/models/artist.rb

```diff
@@ -264,4 +275,44 @@ def k1_blocks
       <block type="simple_jump_left"></block>
     XML
   end
+
+  def strip_block(block)
+    stripped_block = block.dup
+    stripped_block.xpath(STRIPPED_NODES_XPATH).remove
+    stripped_block['type'] = stripped_block['type'].chomp '_dropdown'
+    STRIPPED_ATTRS.each {|attr| stripped_block.remove_attribute(attr)}
+    stripped_block.content = stripped_block.content.strip
+    return stripped_block.to_xml
+  end
+
+  # Add blocks to the toolbox that appear in the solution, but aren't already
+  # in the toolbox
+  def add_missing_toolbox_blocks
+    toolbox = Nokogiri::XML(properties['toolbox_blocks'])
+    toolbox_blocks = toolbox.xpath('//block')
+    Nokogiri::XML(properties['solution_blocks']).xpath('//block').each do |block|
+      next if IGNORED_SOLUTION_BLOCK_ATTRS.any? {|kvpair| block.attr(kvpair[0]) == kvpair[1]}
+
+      stripped_block = strip_block block
+      next if toolbox_blocks.any? do |toolbox_block|
+        stripped_block == strip_block(toolbox_block)
+      end
+
+      # Solution block does not appear in the toolbox, add it
+      toolboxified_block = block.dup
+      toolboxified_block.xpath(STRIPPED_NODES_XPATH).remove
+      toolboxified_block.content = toolboxified_block.content.strip
+      if toolbox.xpath('//category').empty?
+        toolbox.root.add_child toolboxified_block
+      else
+        category = toolbox.xpath('//category[@name=\'NEW BLOCKS\']').first ||
+          toolbox.xpath('//category').last.add_next_sibling(NEW_CATEGORY_XML).first
```

Comment:
I'm hesitant to add an arbitrarily-named category like this, since there is always the chance that a levelbuilder could edit the solution, cause a new block to be added, and then not notice, resulting in "NEW BLOCKS" becoming a user-visible category. I'd like to avoid that.

Unfortunately, I'm not sure what the best solution is. I vote we track down Kiki and find out from her if there's a way to have both categorized and uncategorized blocks in the same toolbox.